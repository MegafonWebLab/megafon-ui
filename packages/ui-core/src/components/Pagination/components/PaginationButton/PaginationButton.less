@import '~styles/base';

@root: e('.mfui-pagination-button');

@{root} {
    .h5();
    display: flex;
    justify-content: center;
    align-items: center;
    width: 32px;
    height: 32px;
    padding: 0;
    border-radius: 50%;
    border: 1px solid transparent;
    background-color: transparent;
    cursor: pointer;

    svg {
        width: 100%;
    }

    &:disabled {
        pointer-events: none;
    }

    &_active {
        border-color: var(--brandGreen);
        pointer-events: none;
    }

    &_theme_default {
        color: var(--content);

        svg {
            fill: var(--content);
        }

        &:hover {
            background-color: var(--spbSky0);
        }
    }

    &_theme_default&_active {
        border-color: var(--brandGreen);
    }

    &_theme_light {
        color: var(--stcWhite);

        svg {
            fill: var(--stcWhite);
        }

        &:hover {
            background-color: var(--stcWhite20);
        }
    }

    &_theme_light&_active {
        border-color: var(--stcWhite);
    }
}
