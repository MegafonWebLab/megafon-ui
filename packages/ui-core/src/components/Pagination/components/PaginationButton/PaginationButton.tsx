import React from 'react';
import { cnCreate, filterDataAttrs } from '@megafon/ui-helpers';
import PropTypes from 'prop-types';
import './PaginationButton.less';

type ValueType = string | number;

export interface IPaginationButtonProps {
    disabled?: boolean;
    isActive?: boolean;
    theme?: 'default' | 'light';
    value?: ValueType;
    className?: string;
    dataAttrs?: {
        root?: Record<string, string>;
    };
    onClick?: (value?: ValueType) => void;
}

const cn = cnCreate('mfui-pagination-button');
const PaginationButton: React.FC<IPaginationButtonProps> = ({
    disabled = false,
    isActive = false,
    theme = 'default',
    className,
    children,
    onClick,
    value,
    dataAttrs,
}) => {
    const handleClick = (): void => {
        onClick?.(value);
    };

    return (
        <button
            {...filterDataAttrs(dataAttrs?.root)}
            className={cn({ active: isActive, theme }, className)}
            disabled={disabled}
            onClick={handleClick}
            type="button"
        >
            {children}
        </button>
    );
};

PaginationButton.propTypes = {
    disabled: PropTypes.bool,
    isActive: PropTypes.bool,
    theme: PropTypes.oneOf(['default', 'light']),
    className: PropTypes.string,
    value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    dataAttrs: PropTypes.shape({
        root: PropTypes.objectOf(PropTypes.string.isRequired),
    }),
    onClick: PropTypes.func,
};

export default PaginationButton;
