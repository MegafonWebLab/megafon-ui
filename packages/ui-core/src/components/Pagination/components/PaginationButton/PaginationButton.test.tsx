import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import PaginationButton from './PaginationButton';

describe('PaginationButton', () => {
    it('should render component', () => {
        const { container } = render(<PaginationButton>Click me</PaginationButton>);

        expect(container).toMatchSnapshot();
    });

    it('should render disabled button', () => {
        const { getByRole } = render(<PaginationButton disabled>Click me</PaginationButton>);

        expect(getByRole('button')).toBeDisabled();
    });

    it('should render active button', () => {
        const { getByRole } = render(<PaginationButton isActive>Click me</PaginationButton>);

        expect(getByRole('button')).toHaveClass('mfui-pagination-button_active');
    });

    it('should render with light theme', () => {
        const { getByRole } = render(<PaginationButton theme="light">Click me</PaginationButton>);

        expect(getByRole('button')).toHaveClass('mfui-pagination-button_theme_light');
    });

    it('should render with className props', () => {
        const { getByRole } = render(<PaginationButton className="testClass">Click me</PaginationButton>);

        expect(getByRole('button')).toHaveClass('testClass');
    });

    it('should render with dataAttrs props', () => {
        const { getByRole } = render(
            <PaginationButton dataAttrs={{ root: { 'data-testid': 'test' } }}>Click me</PaginationButton>,
        );

        expect(getByRole('button')).toHaveAttribute('data-testid', 'test');
    });

    it('should call onClick with value props', () => {
        const value = 1;
        const onClickMock = jest.fn();
        const { getByRole } = render(
            <PaginationButton value={value} onClick={onClickMock}>
                Click me
            </PaginationButton>,
        );

        fireEvent.click(getByRole('button'));

        expect(onClickMock).toBeCalledWith(value);
    });

    it('should call onClick without value', () => {
        const onClickMock = jest.fn();
        const { getByRole } = render(<PaginationButton onClick={onClickMock}>Click me</PaginationButton>);

        fireEvent.click(getByRole('button'));

        expect(onClickMock).toBeCalledWith(undefined);
    });
});
