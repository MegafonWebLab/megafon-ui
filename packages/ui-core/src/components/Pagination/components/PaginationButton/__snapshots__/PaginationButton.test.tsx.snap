// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationButton should render component 1`] = `
<div>
  <button
    class="mfui-pagination-button mfui-pagination-button_theme_default"
    type="button"
  >
    Click me
  </button>
</div>
`;
